src/main/java/org/opensearch/neuralsearch/highlight/SemanticHighlighterEngine.java

```diff
@@ -24,23 +28,60 @@
  * Engine class for semantic highlighting operations
  */
 @Log4j2
+@Builder
+@Value
+@ToString
+@EqualsAndHashCode
 public class SemanticHighlighterEngine {
     private static final String MODEL_ID_FIELD = "model_id";
-    private static final String DEFAULT_PRE_TAG = "<em>";
-    private static final String DEFAULT_POST_TAG = "</em>";
     private static final String MODEL_INFERENCE_RESULT_KEY = "highlights";
     private static final String MODEL_INFERENCE_RESULT_START_KEY = "start";
     private static final String MODEL_INFERENCE_RESULT_END_KEY = "end";
+    private static final String DEFAULT_PRE_TAG = "<em>";
+    private static final String DEFAULT_POST_TAG = "</em>";
+
+    @NonNull
+    MLCommonsClientAccessor mlCommonsClient;
+
+    @NonNull
+    QueryTextExtractorRegistry queryTextExtractorRegistry;
+
+    @Builder.Default
+    String preTag = DEFAULT_PRE_TAG;
+
+    @Builder.Default
+    String postTag = DEFAULT_POST_TAG;
+
+    /**
+     * Creates a new instance with updated highlight tags
+     *
+     * @param newPreTag The new pre tag to use, or null to keep the current one
+     * @param newPostTag The new post tag to use, or null to keep the current one
+     * @return A new SemanticHighlighterEngine instance with the updated tags
+     */
+    public SemanticHighlighterEngine withTags(String newPreTag, String newPostTag) {
+        // Either both tags must be set or both must be null
+        if ((newPreTag == null) != (newPostTag == null)) {
```

Comment:
nit: instead of having multiple newPreTag == null or newPreTag != null check, you can have one boolean flag to store newPreTag != null result, and use the boolean flag in the following check. Same comment for newPostTag. 

Just want to make it more concise.

Reply:
Updated